src/compute-pr-actions.ts

```diff
@@ -394,6 +379,34 @@ export function process(prInfo: PrInfo | BotEnsureRemovedFromProject | BotNoPack
     return context;
 }
 
+function makeStaleness(now: string, author: string, otherOwners: string[]) { // curried for convenience
+    return (kind: StalenessKind, since: Date,
+            freshDays: number, attnDays: number, nearDays: number,
+            doneColumn: ColumnName | "CLOSE") => {
+        const days = daysSince(since, now);
+        const state = days <= freshDays ? "fresh" : days <= attnDays ? "attention" : days <= nearDays ? "nearly" : "done";
+        const kindAndState = `${kind}:${state}`;
+        const explanation = Comments.StalenessExplanations[kindAndState];
+        const comment = Comments.StalenessComment(author, otherOwners)[kindAndState];
+        const doTimelineActions = (context: Actions) => {
+            if (comment !== undefined) {
+                const tag = state === "done" ? kindAndState
+                    : `${kindAndState}:${since.toISOString().replace(/T.*$/, "")}`;
+                context.responseComments.push({ tag, status: comment });
+            }
+            if (state === "done") {
+                if (doneColumn === "CLOSE") {
+                    context.shouldClose = true;
+                    context.shouldRemoveFromActiveColumns = true;
+                } else {
+                    context.targetColumn = doneColumn;
+                }
+            }
+        };
+        return { kind, days, state, explanation, doTimelineActions } as Staleness;

```

Comment:
```suggestion
        return { kind, days, state, explanation, doTimelineActions };

Reply:
I looked at your `state` suggestion, and the inference is done, just not preserved when put in an object.  So adding an `as const` instead of the above achieves the same purpose too, and IMO it's better than both alternatives.

Reply:
Ah that makes sense. Nice :+1: